src/rust/ide/src/ide.rs

```diff
@@ -0,0 +1,184 @@
+    /// as a global executor.
+    ///
+    /// Note: Caller should store or leak this `JsExecutor` so the global
+    /// spawner won't be dangling.
+    pub fn setup_global_executor() -> executor::web::EventLoopExecutor {
+        let executor = executor::web::EventLoopExecutor::new_running();
+        executor::global::set_spawner(executor.spawner.clone());
+        executor
+    }
+
+    /// Establishes transport to the file manager server websocket endpoint.
+    pub async fn connect_to_project_manager(&self) -> Result<WebSocket,ConnectingError> {
+        WebSocket::new_opened(self.logger.clone_ref(),&self.config.project_manager_endpoint).await
+    }
+
+    /// Wraps the transport to the project manager server into the client type and registers it
+    /// within the global executor.
+    pub fn setup_project_manager
+    (transport:impl json_rpc::Transport + 'static) -> project_manager::Client {
+        let project_manager = project_manager::Client::new(transport);
+        executor::global::spawn(project_manager.runner());
+        project_manager
+    }
+
+    /// Creates a new websocket transport and waits until the connection is properly opened.
+    pub async fn new_opened_ws
+    (logger:Logger, address:project_manager::IpWithSocket) -> Result<WebSocket,ConnectingError> {
+        let endpoint = format!("ws://{}:{}", address.host, address.port);
+        WebSocket::new_opened(logger,endpoint).await
+    }
+
+    /// Connect to language server.
+    pub async fn open_project
+    ( logger          : &Logger
+      , json_endpoint   : project_manager::IpWithSocket
+      , binary_endpoint : project_manager::IpWithSocket
+      , project_name    : impl Str
+    ) -> FallibleResult<controller::Project> {
+        info!(logger, "Establishing Language Server connections.");
+        let client_id     = Uuid::new_v4();
+        let json_ws       = Self::new_opened_ws(logger.clone_ref(), json_endpoint).await?;
+        let binary_ws     = Self::new_opened_ws(logger.clone_ref(), binary_endpoint).await?;
+        let client_json   = language_server::Client::new(json_ws);
+        let client_binary = binary::Client::new(logger,binary_ws);
+        crate::executor::global::spawn(client_json.runner());
+        crate::executor::global::spawn(client_binary.runner());
+        let connection_json   = language_server::Connection::new(client_json,client_id).await?;
+        let connection_binary = binary::Connection::new(client_binary,client_id).await?;
+        Ok(controller::Project::new(logger,connection_json,connection_binary,project_name))
+    }
+
+    /// Creates a new project and returns its metadata, so the newly connected project can be
+    /// opened.
+    pub async fn create_project
+    ( logger          : &Logger
+    , project_manager : &impl project_manager::API
+    , name            : &String) -> FallibleResult<ProjectMetadata> {
+        info!(logger, "Creating a new project named `{name}`.");
```

Comment:
Backticks should be used for code. For names use quotes. So replace it with:
`"Creating a new project '{name}'."` instead.